lib/remote/httpserverconnection.cpp

```diff
@@ -542,14 +547,17 @@ void HttpServerConnection::CheckLiveness(boost::asio::yield_context yc)
 	boost::system::error_code ec;
 
 	for (;;) {
-		m_CheckLivenessTimer.expires_from_now(boost::posix_time::seconds(5));
+		// Wait for the half of the liveness timeout to give the connection some leeway to do other work.
+		// But never wait longer than 5 seconds to ensure timely shutdowns.
+		auto sleepTime = std::min(std::chrono::milliseconds(5000), m_LivenessTimeout / 2);
+		m_CheckLivenessTimer.expires_from_now(boost::posix_time::milliseconds(sleepTime.count()));
 		m_CheckLivenessTimer.async_wait(yc[ec]);
 
 		if (m_ShuttingDown) {
 			break;
 		}
 
-		if (m_Seen < Utility::GetTime() - 10) {
+		if (m_Seen < std::chrono::system_clock::now() - m_LivenessTimeout) {
```

Comment:
I think `steady_clock` would actually make more sense here. Sure, `system_clock` better models what `Utility::GetTime()` did so far, but `m_Seen` seems to be used only for this check and this only cares about how much time has passed, not about what the actual time is. `system_clock` may jump, so if you set the system time forward by a minute, that would trigger this condition even if the client wasn't idle for that long.